tls/extensions/s2n_client_key_share.c

```diff
@@ -20,7 +20,11 @@
 #include "error/s2n_errno.h"
 #include "stuffer/s2n_stuffer.h"
 #include "utils/s2n_safety.h"
+#include "tls/s2n_tls13.h"
 
+#define S2N_GENERATE_EMPTY_KEY_SHARE_LIST(preferred_key_shares) (preferred_key_shares & 1)
+#define S2N_GENERATE_KEY_SHARE_FOR_SELECTED_GROUP(preferred_key_shares, i) ((preferred_key_shares >> (i + 1)) & 1)
```

Comment:
These names makes it sound like these macros are actually doing the generation. I would aim for something with "is" in the title, to indicate it's a boolean check.

"s2n_is_key_share_list_empty"? "is_key_share_requested"?

Reply:
Yeaah, naming is hard :/